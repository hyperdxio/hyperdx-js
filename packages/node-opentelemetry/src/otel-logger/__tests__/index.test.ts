import { getSeverityNumber, parseLogAttributes } from '..';

describe('otel-logger', () => {
  it('getSeverityNumber', () => {
    expect({
      alert: getSeverityNumber('alert'),
      debug: getSeverityNumber('debug'),
      emerg: getSeverityNumber('emerg'),
      error: getSeverityNumber('error'),
      info: getSeverityNumber('info'),
      verbose: getSeverityNumber('verbose'),
      warn: getSeverityNumber('warn'),
      warning: getSeverityNumber('warning'),
    }).toMatchSnapshot();
  });

  it('parseLogAttributes', () => {
    expect(
      parseLogAttributes({
        a: 1,
        b: '2',
        c: [{ d: 3 }],
        e: [1, 2, 3],
        f: ['a', 'b', 'c'],
        g: {
          h: 'i',
        },
      }),
    ).toMatchSnapshot();
  });
});
