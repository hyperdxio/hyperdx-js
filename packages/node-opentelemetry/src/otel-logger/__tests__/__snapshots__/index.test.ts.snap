// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`otel-logger getSeverityNumber 1`] = `
{
  "alert": 22,
  "debug": 5,
  "emerg": 23,
  "error": 17,
  "info": 9,
  "verbose": 6,
  "warn": 13,
  "warning": 13,
}
`;

exports[`otel-logger parseLogAttributes 1`] = `
{
  "a": 1,
  "b": "2",
  "c": "[{"d":3}]",
  "e": [
    1,
    2,
    3,
  ],
  "f": [
    "a",
    "b",
    "c",
  ],
  "g": {
    "h": "i",
  },
}
`;
